src/cursor/cursor.ts

```diff
@@ -0,0 +1,121 @@
+import * as proto from "google-protobuf";
+import { ClientReadableStream } from "@grpc/grpc-js";
+import { TigrisCursorInUseError } from "../error";
+
+/** @internal */
+export interface Initializer<TResp extends proto.Message> {
+	init(): ClientReadableStream<TResp>;
+}
+
+/** @internal */
+const tStream = Symbol("stream");
+/** @internal */
+const tReady = Symbol("ready");
+/** @internal */
+const tClosed = Symbol("closed");
+
+export abstract class Cursor<TSchema, TResp extends proto.Message> {
```

Comment:
we can extend this interface in future by taking motivation from nicely design Java's stream interface https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/util/stream/Stream.html